core/src/main/java/org/hisp/dhis/android/core/arch/d2/internal/D2DIComponentFactory.kt

```diff
@@ -75,70 +75,86 @@ import org.hisp.dhis.android.core.validation.ValidationDIModule
+                TrackerDIModule().module,
+                UseCaseDIModule().module,
+                UserDIModule().module,
+                ValidationDIModule().module,
+                VisualizationDIModule().module,
+                WipeDIModule().module,
+            ),
+        )
+    }
+
+    lateinit var koin: Koin
+}
+
+internal interface IsolatedKoinComponent : KoinComponent {
+    override fun getKoin(): Koin = DhisAndroidSdkKoinContext.koin
+}
 
-                    AnalyticsDIModule().module,
-                    ArchDIModule().module,
-                    AttributeDIModule().module,
-                    CategoryDIModule().module,
-                    CommonDIModule().module,
-                    ConfigurationDIModule().module,
-                    ConstantDIModule().module,
-                    DataApprovalDIModule().module,
-                    DataElementDIModule().module,
-                    DataSetDIModule().module,
-                    DataStoreDIModule().module,
-                    DataValueDIModule().module,
-                    DomainDIModule().module,
-                    EnrollmentDIModule().module,
-                    EventDIModule().module,
-                    ExpressionDimensionItemDIModule().module,
-                    FileResourceDIModule().module,
-                    IconDIModule().module,
-                    ImportsDIModule().module,
-                    IndicatorDIModule().module,
-                    LegendSetDIModule().module,
-                    MaintenanceDIModule().module,
-                    MapDIModule().module,
-                    NoteDIModule().module,
-                    NetworkDIModule().module,
-                    OptionDIModule().module,
-                    OrganisationUnitDIModule().module,
-                    ParserDIModule().module,
-                    PeriodDIModule().module,
-                    ProgramDIModule().module,
-                    ProgramStageWorkingListDIModule().module,
-                    RelationshipDIModule().module,
-                    ResourceDIModule().module,
-                    SettingsDIModule().module,
-                    SmsDIModule().module,
-                    SystemInfoDIModule().module,
-                    TrackedEntityDIModule().module,
-                    TrackerDIModule().module,
-                    UseCaseDIModule().module,
-                    UserDIModule().module,
-                    ValidationDIModule().module,
-                    VisualizationDIModule().module,
-                    WipeDIModule().module,
-                ),
-            )
-        }
+internal object D2DIComponentFactory : IsolatedKoinComponent {
 
-        return koinApp.koin.get<D2DIComponent>()
+    fun create(
+        d2Config: D2Configuration,
+        secureStore: SecureStore,
+        insecureStore: InsecureStore,
+    ): D2DIComponent {
+        DhisAndroidSdkKoinContext.koin = DhisAndroidSdkKoinContext.create(d2Config, secureStore, insecureStore).koin
```

Comment:
I find this part a bit odd... It is assigning a property in an object and then using method declared in an interface (`getKoin()`), which internally refers to the same property in the that object. Couldn't it be simplified in some way? Or abstracted in some way?